src/cryptography/nostr_secure_rng.cpp

```diff
@@ -0,0 +1,46 @@
+#include <plog/Init.h>
+#include <plog/Log.h>
+
+#include <openssl/evp.h>
+#include <openssl/rand.h>
+#include <openssl/crypto.h>
+
+#include "nostr_secure_rng.hpp"
+
+using namespace std;
+using namespace nostr::cryptography;
+
+void NostrSecureRng::fill(void* buffer, size_t length)
```

Comment:
I don't think we want any type of "generics" here because the underlying system operates on memory directly so it's only one type `uint8_t*` so we could change `void*` to `uint8_t8`. I wouldn't be opposed to some overloads that handle the type conversion and buffer sizing, but I don't really see a use case atm.

Reply:
Makes sense to me.  If we're only using `uint8_t`, there's no need for generics.